scriptworker/task.py

```diff
@@ -250,6 +250,29 @@ def guess_content_type(path):
                     )
 
 
+def _compress_if_text_plain(artifact_path, content_type):
+    if content_type == 'text/plain':
+        log.debug('"{}" is a plain text file. Compressing...'.format(artifact_path))
+        with open(artifact_path, 'rb') as f_in:
+            text_content = f_in.read()
+
+        with gzip.open(artifact_path, 'wb') as f_out:
+            f_out.write(text_content)
+        log.info('Compressed "{}" with gzip'.format(artifact_path))
+    else:
+        log.debug('"{}" is NOT a plain text file, nothing to compress.'.format(artifact_path))
+
+
+def _craft_artifact_put_hearders(tc_response):
+    headers = {
+        aiohttp.hdrs.CONTENT_TYPE: tc_response['contentType'],
+    }
+    if tc_response['contentType'] == 'text/plain':
```

Comment:
I agree, `guess_content_type` is a better central place for handling content-type and encoding. While testing [the arguments above](https://github.com/mozilla-releng/scriptworker/pull/53#discussion_r95971790), I realized I had to change one how a given content-type is handled at several places.